src/modules/change-state.js

```diff
@@ -178,19 +178,22 @@
 		}
 		
 		var followers = followerScope.find(followerSelector);
+		var notifyData = {item: item, state: state, flag: flag, cancel: false};
 
-		App.modules.notify('changeState.begin', {item: item, state: state, flag: flag});
+		App.mediator.notify('changeState.begin', notifyData);
 
 		//Execute change
-		doSetItemState(item, state, flag);
+		if (!!!notifyData.cancel && !notifyData.cancel) {
+			doSetItemState(item, state, flag);
 
-		//Process followers
-		followers.each(function () {
-			var it = $(this);
-			setItemState(it, state, flag);
-		});
-		
-		App.modules.notify('changeState.end', {item: item, state: state, flag: flag});
+			//Process followers
+			followers.each(function () {
+				var it = $(this);
+				setItemState(it, state, flag);
+			});
+		}
+
+		App.modules.notify('changeState.end', {item: item, state: state, flag: flag, canceled: notifyData.cancel});
```

Comment:
Should we pass `notifyData` completely here ? One could "store" values in the begin and get them in the end... (?)